### DISCUSSION TITLE
Add-On Not Starting After New Install

### DISCUSSION THREAD

**Discussion by JoeFromQA** [tsightler/ring-mqtt#503] [Support Q&A]
Hello,

I just installed the Ring-MQTT for my Ring Alram Gen 2 system (No cameras). When I click to start the addon, nothing happens, then I get the following message in the log:

Running ring-mqtt...
2022-11-17T23:44:18.798Z ring-mqtt Detected runmode: addon
2022-11-17T23:44:18.800Z ring-mqtt Configuration file: /data/options.json
2022-11-17T23:44:18.803Z ring-mqtt Discovered MQTT Host: core-mosquitto
2022-11-17T23:44:18.803Z ring-mqtt Discovered MQTT Port: 1883
2022-11-17T23:44:18.804Z ring-mqtt Discovered MQTT User: addons
2022-11-17T23:44:18.804Z ring-mqtt Discovered MQTT password: <hidden>
2022-11-17T23:44:18.804Z ring-mqtt MQTT URL: mqtt://addons:********@core-mosquitto:1883
**2022-11-17T23:44:19.917Z ring-mqtt State file /data/ring-state.json not found. No saved state data available.**
2022-11-17T23:44:19.918Z ring-mqtt Succesfully started the token generator web UI
**2022-11-17T23:44:19.919Z ring-mqtt No refresh token was found in state file, generate a token using the addon Web UI**

Watching someone on [youtube ](https://youtu.be/GXUKW9WMI2c?t=256) set this up, I see the Web UI option doesnt show until _after_  they start the addon, so I am not sure where to go from here to get the token generated.  

I have tried reinstalling the addon a few times and restarting HA which seemed to fix a similar issue for others I found, but no luck. 

Any help would be appreciated. Thanks!! 

Running the following:
Home Assistant 2022.11.3 
Supervisor 2022.10.2
Operating System 9.3
Frontend 20221108.0 - latest

Mosquitto broker
Current version: 6.1.3

Ring-MQTT with Video Streaming
Current version: 5.0.5

**Comment by tsightler** [comment]
Everything above is normal if this is the first time starting the addon, just click the "Open Web UI" button on the addon Info tab and authenticate.

**Reply by JoeFromQA** [reply]
Thanks for the quick reply!! 

That is the issue, when I click start I get a very quick circle over the start button, but then nothing happens. The add-on does not start, and the "Open Web UI" button does not show. 

Here is what I am seeing:

![start](https://user-images.githubusercontent.com/22438477/202600946-dbcf44aa-e5dd-42e8-856c-169afb65d5cb.gif)

Anything else I should look into?

Thanks for your help!

**Reply by tsightler** [reply]
But the logs clearly show that the addon has started.  Wait about 5-10 seconds after clicking start and then just force a browser refresh instead of just waiting for it to update.

**Reply by JoeFromQA** [reply]
I tried on other browsers and even using the mobile app and got the same "spin and nothing". I tried restarting HA in the UI many times in between with no luck.

**I then restarted the whole VM HA is hosted on, and that did the trick!** 

Looking forward to using my Ring Alarm in HA now! Thanks @tsightler !!

**Reply by tsightler** [reply]
Must have been some strange quirk with the HA UI, I've never seen anything like that behavior.  Glad you got it working!

**Reply by Samirs1000** [reply]
I'm having the exact same issue. I have reinstalled the mqtt and the live-service addon will not start anymore and so none of my sensors are being picked up anymore. Unfortunately restarting both HA and my laptop have not fixed the issue. Any other solutions for how to get the add-on to start?

**Reply by tsightler** [reply]
What do the logs say?  How do you know the addon is not starting?  If the addon is truly not starting then this is a Home Assistant issue but the Home Assistant Supervisor logs will tell you.  If the addon is starting but you only get messages in the addon logs about invalid token (which is what will happen if you re-install) then you just have to re-authenticate.

**Reply by Samirs1000** [reply]
The logs said it was starting but like the above post the UI didn't show it starting so I couldn't authenticate. I managed to restore an older version, updated the token then restored my most recent back up so all sorted now. Thanks
